### PR TITLE
My pull Request

### PR DESCRIPTION
**Pull request by mondris360**

### CODE REVIEW

**Comment by mondris360** [comment]
Hello Bond, i am sorry for submitting this after the deadline. Please help me to  review it

File: src/binary-reversal/index.js [hunk 1]
```diff
@@ -3,6 +3,31 @@
...
  */
-function binaryReversal(value) {}
+function binaryReversal(value) {
+    const integerValue = Number(value);
+    // convert to binary
+    const binaryStr = integerValue.toString(2);
+     // calculate the number of zeroes need to pad the string
+    const noOfZeroesToAdd = (binaryStr.length % 8) == 0 ? 0 : 8 -  (binaryStr.length % 8);
+     // pad the string
+     const paddedString =  "0".repeat(noOfZeroesToAdd) + binaryStr;
```
Comment (bondz) [inline src/binary-reversal/index.js:13]: This is equivalent to `binaryStr.padStart(8, 0)`;

File: src/binary-reversal/index.js [hunk 2]
```diff
@@ -3,6 +3,31 @@
...
+    const reversedStr =  reverseString(paddedString);
```
Comment (bondz) [inline src/binary-reversal/index.js:14]: You can `split` and `reverse` and `join`.

File: src/binary-reversal/index.js [hunk 3]
```diff
@@ -3,6 +3,31 @@
...
+    // convert the reversed string to decimal
+    const binaryToDecimal =  parseInt(reversedStr, 2);
+    const answer = binaryToDecimal.toString();
+    
+    return answer;
```
Comment (bondz) [inline src/binary-reversal/index.js:19]: You can also just return without creating that extra variable.

File: src/list-sorting/index.js [hunk 4]
```diff
@@ -1,3 +1,20 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+    const arrayLength =  haystack.length;
+    for (let x= arrayLength-1; x>=0; x--){
+      // if the current element is an array
+        nestedArr = typeof(haystack[x]) == 'object';
```
Comment (bondz) [inline src/list-sorting/index.js:5]: Prefer to use `Array.isArray` to check if a variable is an array.

File: src/list-sorting/index.js [hunk 5]
```diff
@@ -1,3 +1,20 @@
...
+        if(nestedArr) {
+            let start = haystack[x].length-1;
+            for (y = start; y>=0; y--){
+                if(haystack[x][y] == needle) return [x,y];
+            }
```
Comment (bondz) [inline src/list-sorting/index.js:10]: This is essentially `[].lastIndexOf`

File: src/list-sorting/index.js [hunk 6]
```diff
@@ -1,3 +1,20 @@
-function listSorting(needle, haystack) {}
+function listSorting(needle, haystack) {
+    const arrayLength =  haystack.length;
+    for (let x= arrayLength-1; x>=0; x--){
```
Comment (bondz) [inline src/list-sorting/index.js:3]: `x` could be `row`

**Pull request merged**